### PR TITLE
Re-calculate duration in split-out workspaces from FilterEvents

### LINKED ISSUE #27677: FilterEvents shall re-calculate duration
### Expected behavior

Log value 'duration' shall be re-calculated for each split-out workspace from FilterEvents

### Actual behavior

Not doing so.

### PR DESCRIPTION
**Pull request by wdzhou**
**Description of work.**

Re-calculate the duration, as a property in run, of each output (split) workspace.

**To test:**

1. Review the expanded unit test
1. Load a workspace, generate splitter manually with known duration, call FilterEvents and then check the sample log 'duration' from all the output workspaces.

Fixes #27677 .

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/FilterEvents.cpp [hunk 1]
```diff
@@ -185,6 +186,12 @@ void FilterEvents::init() {
   declareProperty("DescriptiveOutputNames", false,
                   "If selected, the names of the output workspaces will "
                   "include information about each slice.");
+
```
Comment (peterfpeterson) [inline Framework/Algorithms/src/FilterEvents.cpp]: Remove this commented out code.

File: Framework/Algorithms/src/FilterEvents.cpp [hunk 2]
```diff
@@ -1964,45 +1971,78 @@ void FilterEvents::generateSplitterTSPalpha(
...
+    // std::move()
+    outws->mutableRun().addProperty(std::move(split_tsp_vec[itarget]), true);
+  }
 
   return;
 }
 
+/** Calculate split-workspace's duration according to splitter time series
+ * property
+ * @brief FilterEvents::calculate_duration
```
Comment (peterfpeterson) [inline Framework/Algorithms/src/FilterEvents.cpp]: Remove unhelpful doxygen, or fill in the information

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
Remove some commented out code and add release notes.

File: Framework/Algorithms/src/FilterEvents.cpp (hunk 1 above)
Comment (wdzhou) [reply Framework/Algorithms/src/FilterEvents.cpp]: Done

File: Framework/Algorithms/src/FilterEvents.cpp (hunk 2 above)
Comment (wdzhou) [reply Framework/Algorithms/src/FilterEvents.cpp]: Better @brief

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**